cli/wrapper.go

```diff
@@ -0,0 +1,418 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cli
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http/httptest"
+	"strings"
+
+	blockstore "github.com/ipfs/boxo/blockstore"
+	"github.com/libp2p/go-libp2p/core/peer"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/datastore"
+	"github.com/sourcenetwork/defradb/events"
+	"github.com/sourcenetwork/defradb/http"
+)
+
+var _ client.DB = (*Wrapper)(nil)
+
+type Wrapper struct {
```

Comment:
question: Why is this production code?

Reply:
I'd like to move this and the http wrapper to the integration tests package. I think there was something blocking me from doing that, but I will give it another look.

Reply:
Cheers Keenan - please give me a shout here once you've pushed changes/decided against moving and I'll review them fully.

Reply:
Update: I was able to move them to a new `tests/clients` package.